src/index.js

```diff
@@ -119,12 +132,15 @@ class LumigoPlugin {
 		}
 	}
 
-	getFunctionsToWrap(service) {
-		const functions = service.getAllFunctions().map(localName => {
-			const x = _.cloneDeep(service.getFunction(localName));
-			x.localName = localName;
-			return x;
-		});
+	getFunctionsToWrap(service, functionNames) {
+		const functions = service
+			.getAllFunctions()
+			.filter(localName => !functionNames || functionNames.includes(localName))
```

Comment:
I prefer to avoid `undefined` check, just pass everything if you want all functions or the specific function. It's confusing

Reply:
I wanted to keep `functionNames` as an optional parameter.  
Passing `service.getAllFunctions()` names just to filter out any of those that aren't in that same set may subjectively be even more confusing.  

I have pushed [this change](https://github.com/lumigo-io/serverless-lumigo-plugin/pull/24/commits/87e2e82f496b5e5cec962d447ecf8219d1469fd1), Do you find it less confusing now?